OrbitBase/ThreadUtilsLinux.cpp

```diff
@@ -22,13 +23,16 @@ pid_t GetCurrentThreadId() {
 }
 
 std::string GetThreadName(pid_t tid) {
-  char thread_name[kMaxThreadNameLength];
-  int result = pthread_getname_np(pthread_self(), thread_name, kMaxThreadNameLength);
-  if (result != 0) {
-    ERROR("Getting thread name for tid %d. Error %d", tid, result);
-    return {};
+  std::string comm_filename = absl::StrFormat("/proc/%d/comm", tid);
+  ErrorMessageOr<std::string> comm_content = ReadFileToString(comm_filename);
+  if (!comm_content.has_value()) {
+    ERROR("Getting thread name for tid %d: %s", tid, comm_content.error().message());
+    return "";
+  }
+  if (comm_content.value().back() == '\n') {
```

Comment:
absl::StripSuffix? It is slower though....

Reply:
also - what if it is empty.. also not sure if this could happen, can you set "" name for a thread?

Reply:
also this relies on the fact that value() returns a reference. maybe add auto value = std::move(comm_content.value()) and use that here and below?

Reply:
I don't mind it, but indeed it copies the string. Up to you.

Reply:
> I don't mind it, but indeed it copies the string. Up to you.

I would keep it as it is. the '\n' check...

Reply:
> also this relies on the fact that value() returns a reference. maybe add auto value = std::move(comm_content.value()) and use that here and below?

No, it's normal that `value` returns a reference, it would be quite a weird interface otherwise.

Reply:
> also - what if it is empty.. also not sure if this could happen, can you set "" name for a thread?

Done.

Reply:
I would still add a check for value.empty(), I am not sure if the test works because comm always has \n at the end or we are just lucky.

Reply:
Aaah, you are right, I didn't understand what you meant above, this worked because the `\n` was always there but if we have the `if` we need to do it right! Added `!comm_content.value().empty() &&` to the condition.

Reply:
Tell me if it looks good now before I get it wrong again.